lib/_stream_wrap.js

```diff
@@ -47,12 +44,12 @@ function StreamWrap(stream) {
     self.emit('error', err);
   });
   this.stream.on('data', function ondata(chunk) {
-    if (!(chunk instanceof Buffer)) {
+    if (typeof chunk === 'string' || this._readableState.objectMode === true) {
       // Make sure that no further `data` events will happen
       this.pause();
       this.removeListener('data', ondata);
 
-      self.emit('error', new errors.Error('ERR_STREAM_HAS_STRINGDECODER'));
+      self.emit('error', new errors.Error('ERR_STREAM_WRAP'));
```

Comment:
Given that I don't believe the other error code had actually shipped in a release yet, this is fine. Definitely makes this semver-major tho.